src/main/scala/firrtl/transforms/BlackBoxSourceHelper.scala

```diff
@@ -67,40 +67,33 @@ class BlackBoxSourceHelper extends firrtl.Transform {
     */
   override def execute(state: CircuitState): CircuitState = {
     val (annos, targetDir) = collectAnnos(state.annotations)
-    val fileList = annos.foldLeft(List.empty[String]) {
-      case (fileList, anno) => anno match {
-        case BlackBoxResourceAnno(_, resourceId) =>
-          val name = resourceId.split("/").last
-          val outFile = new File(targetDir, name)
-          BlackBoxSourceHelper.copyResourceToFile(resourceId,outFile)
-          outFile.getAbsolutePath +: fileList
-        case BlackBoxInlineAnno(_, name, text) =>
-          val outFile = new File(targetDir, name)
-          val writer = new PrintWriter(outFile)
-          writer.write(text)
-          writer.close()
-          outFile.getAbsolutePath +: fileList
-        case _ => throwInternalError()
-      }
+
+    val resourceFiles: Set[File] = annos.collect {
+      case BlackBoxResourceAnno(_, resourceId) =>
+        val name = resourceId.split("/").last
+        val outFile = new File(targetDir, name)
+        (resourceId, outFile)
+    }.toSet[(String, File)].map { case (res, file) =>
```

Comment:
I actually think it would be better to use `.distinct` here instead of changing to a `Set` just for determinism in the ordering